tool-openssl/ec.cc

```diff
@@ -0,0 +1,115 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#include <openssl/bio.h>
+#include <openssl/ec.h>
+#include <openssl/err.h>
+#include <openssl/pem.h>
+#include <string>
+#include "internal.h"
+
+enum Format {
+  FORMAT_PEM = 1,
+  FORMAT_DER = 2
+};
+
+static const argument_t kArguments[] = {
+    {"-help", kBooleanArgument, "Display this summary"},
+    {"-inform", kOptionalArgument, "Input format (PEM or DER), default PEM"},
+    {"-in", kOptionalArgument, "Input file, default stdin"},
+    {"-pubout", kBooleanArgument, "Output public key, not private"},
+    {"-out", kOptionalArgument, "Output file, default stdout"},
+    {"-outform", kOptionalArgument, "Output format (PEM or DER), default PEM"},
+    {"", kOptionalArgument, ""}};
+
+bool ecTool(const args_list_t &args) {
+  ordered_args::ordered_args_map_t parsed_args;
+  args_list_t extra_args;
+  std::string in_path, out_path, inform_str, outform_str;
+  bool help = false, pubout = false;
+  int input_format = FORMAT_PEM, output_format = FORMAT_PEM;
+  bssl::UniquePtr<BIO> input_bio, output_bio;
+  bssl::UniquePtr<EC_KEY> ec_key;
+
+  if (!ordered_args::ParseOrderedKeyValueArguments(parsed_args, extra_args,
+                                                   args, kArguments)) {
+    PrintUsage(kArguments);
+    goto err;
+  }
+
+  ordered_args::GetBoolArgument(&help, "-help", parsed_args);
+  ordered_args::GetString(&in_path, "-in", "", parsed_args);
+  ordered_args::GetString(&out_path, "-out", "", parsed_args);
+  ordered_args::GetString(&inform_str, "-inform", "PEM", parsed_args);
+  ordered_args::GetString(&outform_str, "-outform", "PEM", parsed_args);
+  ordered_args::GetBoolArgument(&pubout, "-pubout", parsed_args);
+
+  if (help) {
+    PrintUsage(kArguments);
+    return true;
+  }
+
+  if (inform_str == "PEM" || inform_str == "pem") {
```

Comment:
Use `isStringUpperCaseEqual` as you have for the "ecparam" command.

A "parse" method for this could even be included in the `Format` type.